handlers/books_handler.go

```diff
@@ -0,0 +1,61 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"main/repository"
+	"net/http"
+	"sort"
+)
+
+func HandleBookRequests(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		getBooks(w, r)
+	} else if r.Method == http.MethodPost {
+		postBooks(w, r)
+	} else {
+		fmt.Println("supporting only POST and GET requests")
+	}
+}
+
+func getBooks(w http.ResponseWriter, r *http.Request) {
+	booksRepo := repository.NewBookRepo()
```

Comment:
don't create repo here, create a struct and assign your funcs to the struct, then add BookRepo parameter to the struct and reuse it in every method, also you are going to name the method from Big letter so it will be able to export